internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -1791,7 +1791,12 @@ func resourceKubernetesClusterRead(d *pluginsdk.ResourceData, meta interface{})
 		d.Set("private_fqdn", props.PrivateFQDN)
 		d.Set("portal_fqdn", props.AzurePortalFQDN)
 		d.Set("disk_encryption_set_id", props.DiskEncryptionSetID)
+		// CurrentKubernetesVersion contains the actual version the Managed Cluster is running after upgrading
+		// KubernetesVersion now seems to contain the initial version the cluster was created with
 		d.Set("kubernetes_version", props.KubernetesVersion)
+		if v := props.CurrentKubernetesVersion; v != nil {
```

Comment:
given this has changed, does the node pool rotation logic need to changed?

Reply:
Upon closer inspection the logic does not need to be changed, but for the case where the version we are upgrading from is not supported by the control plane we do need to get the value from `CurrentKubernetesVersion` for the error message.